internal/uuid/uuidutil_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package uuid
+
+import (
+	"sync"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/internal/randutil"
+	"go.mongodb.org/mongo-driver/internal/uuid/uuid"
+
+	"github.com/stretchr/testify/require"
+)
+
+// GODRIVER-2349
+// Test that initializing many package-global UUID sources concurrently never leads to any duplicate
+// UUIDs being generated.
+func TestGlobalSource(t *testing.T) {
+	t.Run("exp rand 1 UUID x 1,000,000 goroutines", func(t *testing.T) {
+		// Read a UUID from each of 1,000,000 goroutines and assert that there is never a duplicate value.
+		uuid.SetRand(randutil.GlobalRand)
+		uuids := new(sync.Map)
+		var wg sync.WaitGroup
+		for i := 1; i < 1000000; i++ {
+			wg.Add(1)
```

Comment:
```suggestion 
                iterations := 1e6
		wg := sync.WaitGroup{}.Add(iterations)
		for i := 1; i < threshold; i++ {
```
I think this will help ensure this subtest is consistent with the other subtests.  I think decoupling `Add` from the loop will also make the test safer, although it's probably fine as is.

Reply:
Good point! Thank you!